source/synthDrivers/sapi5.py

```diff
@@ -396,18 +340,60 @@ def outputTags():
 
 		text = "".join(textList)
 		flags = SpeechVoiceSpeakFlags.IsXML | SpeechVoiceSpeakFlags.Async
-		self.tts.Speak(text, flags)
+		# Although background audio is ducked in SAPISink.StartStream
+		# and unducks in SAPISink.EndStream,
+		# those events are asynchronous and therefor enabling audio ducking
+		# cannot enforce a delay before the speech stream starts,
+		# while audio is being faded down.
+		# Therefore, create a temporary AudioDucker object,
+		# and enable and disable it directly around the call to speak,
+		# So that enabling audio ducking suitably delays the speak call if necessary.
+		# Although speak does not block and audio ducking is disabled straight after,
+		# Audio will be still ducked for enough time for the subsequent StartStream to keep it enabled.

```

Comment:
Rather than use a tempAudioDucker, why not give `speak` the responsibility of enabling audio ducking, and `StropStream` the responsibility of disabling. Why is `StartStream` needed?

Reply:
When calling speak, it is possible that the synthesizer may not produce any audio, for instance if the text is empty or does not result in any pronounceable symbols. In that case, StartStream and EndStream will never be fired. This would be a very rare situation, but certainly not impossible. Really StartStream and EndStream are the true accurate places for controlling ducking. But because they are fired asynchronously, the enforced blocking when calling AudioDucker.enable (when not already ducked) will not block the start of speech, thus why we also need to do it temporarily around speak, to ensure that speak is delayed enough.

Reply:
Does temp audio ducker actually cause the audio to be ducked, or is it only used for timing?